router/downloader/downloader.go

```
@@ -163,6 +163,17 @@ func (dl *Download) Execute() error {
 		return ErrDownloadFailed
 	}
 	defer res.Body.Close()
+
+	//If the http code is a redirect get the location and set the url to that
```

Please note the security implications of this (noted at the start): https://github.com/pterodactyl/wings/blob/3b0ecca1c4f01ef558927d528a82083011b2d060/router/downloader/downloader.go#L25-L33

---

Since it uses recursion and the execute function first checks if the URL is in the local network it should throw an error if its in the local network.